lib/buffer.js

```diff
@@ -247,6 +247,11 @@ Buffer.concat = function(list, length) {
   if (!Array.isArray(list))
     throw new TypeError('list argument must be an Array of Buffers.');
 
+  if (list.length === 0)
+    return new Buffer(0);
+  else if (list.length === 1)
+    return list[0];
```

Comment:
By the way, should we follow the below assertions:

```
assert.equal(Buffer.cancat(anyArray, len).length, len);
assert.equal(Buffer.cancat([], 1).length, 1); // this line would fail
assert.equal(Buffer.concat([new Buffer(20)]).length, 10); this line would fail as well and importantly i think we need to cut the buffer to 10.
```

@JacksonTian I know this should not be related to your patch, but it would be great if this is able to be fixed(bug? unsure) :p though cc @trevnorris